main/FatesHistoryInterfaceMod.F90

```diff
@@ -4861,7 +4660,307 @@ subroutine update_history_dyn2(this,nc,nsites,sites,bc_in)
+         hio_fire_sum_fuel_si_age           => this%hvars(ih_fire_sum_fuel_si_age)%r82d, &
+         hio_fuel_amount_age_fuel            => this%hvars(ih_fuel_amount_age_fuel)%r82d, &
+!         hio_fire_rate_of_spread_front_si_age => this%hvars(ih_fire_rate_of_spread_front_si_age)%r82d, &
+         hio_mortality_canopy_si_scag         => this%hvars(ih_mortality_canopy_si_scag)%r82d, &
+         hio_mortality_understory_si_scag     => this%hvars(ih_mortality_understory_si_scag)%r82d, &
+         hio_biomass_si_age        => this%hvars(ih_biomass_si_age)%r82d, &
+         hio_biomass_si_agepft                => this%hvars(ih_biomass_si_agepft)%r82d, &
+         hio_npp_si_agepft                    => this%hvars(ih_npp_si_agepft)%r82d, &
+         hio_ddbh_canopy_si_scag              => this%hvars(ih_ddbh_canopy_si_scag)%r82d, &
+         hio_fire_intensity_si_age          => this%hvars(ih_fire_intensity_si_age)%r82d, &
+         hio_npatches_si_age                  => this%hvars(ih_npatches_si_age)%r82d, &
+         hio_canopy_fracarea_si_age           => this%hvars(ih_canopy_fracarea_si_age)%r82d, &
+         hio_nplant_si_scag                   => this%hvars(ih_nplant_si_scag)%r82d, &
+         hio_nplant_si_scagpft                => this%hvars(ih_nplant_si_scagpft)%r82d, &
+         hio_nplant_canopy_si_scag            => this%hvars(ih_nplant_canopy_si_scag)%r82d, &
+         hio_nplant_understory_si_scag        => this%hvars(ih_nplant_understory_si_scag)%r82d, &
+         hio_fracarea_si_age                  => this%hvars(ih_fracarea_si_age)%r82d, &
+         hio_agesince_anthrodist_si_age       => this%hvars(ih_agesince_anthrodist_si_age)%r82d, &
+         hio_secondarylands_fracarea_si_age   => this%hvars(ih_secondarylands_fracarea_si_age)%r82d, &
+         hio_ddbh_understory_si_scag          => this%hvars(ih_ddbh_understory_si_scag)%r82d)
+
+    siteloop: do s = 1,nsites
+       io_si  = sites(s)%h_gid
+
+       ! Loop through patches to sum up diagnostics
+       cpatch => sites(s)%oldest_patch
+       patchloop: do while(associated(cpatch))
+          cpatch%age_class  = get_age_class_index(cpatch%age)
+          patch_area_div_site_area = cpatch%area * AREA_INV
+          patch_canarea_div_site_area = cpatch%total_canopy_area * AREA_INV
+
+          !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+          !!! Weighting by (or using) patch area relative to total site area !!!
+          !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+          ! Increment the fractional area in each age class bin
+          hio_fracarea_si_age(io_si,cpatch%age_class) = hio_fracarea_si_age(io_si,cpatch%age_class) &
+          + cpatch%area * AREA_INV
+
+          do ft = 1,numpft
+             iagepft = get_agepft_class_index(cpatch%age,ft)
+             hio_scorch_height_si_pft(io_si,ft) = hio_scorch_height_si_pft(io_si,ft) + &
+                  cpatch%Scorch_ht(ft) * patch_area_div_site_area
```

Comment:
Wait... Why is this in `update_history_dyn_subsite_ageclass()`? It's just a PFT-dimensioned variable.

Reply:
Same for `hio_zstar_si` and `hio_fuel_amount_age_fuel`.

Reply:
good point, I'll move these

Reply:
Nah, I will, after I'm done with the merging (almost there).

- [ ] Fix this.

Reply:
Resolved with 4e91481db and c158e46cd.